ambry-router/src/main/java/com.github.ambry.router/SimpleOperationTracker.java

```diff
@@ -238,9 +243,29 @@
         routerOperation, successTarget, parallelism, originatingDcNotFoundFailureThreshold);
   }
 
+  /**
+   * The dynamic success target is introduced mainly for following use case:
+   * In the intermediate state of "move replica", when decommission of old replicas is initiated(but hasn't transited to
```

Comment:
Does this apply to the scenario when adding replicas to the partition? When it's bootstrapping, ambry server would return Temporary_Disable to put request as well.

Reply:
Good question, we actually don't have to worry about new replicas that are bootstrapping because from frontends' point view, they are either in OFFLINE or BOOTSTRAP state which won't be picked for PUT operation. (For PUT, the eligible states are STANDBY and LEADER)
The problem addressed by this PR is because frontends are still seeing old view in which some replicas are still STANDBY/LEADER but their actual state is being decommissioned.